scripts/pre_push_hook.py

```diff
@@ -59,6 +59,9 @@
 
 
 class ChangedBranch(object):
+    """Class for changing branch in which modified file is present by passing
```

Comment:
I think you should explain that this is a context manager class (please look up what that is). Normally a class doesn't "do" things so I was confused when I first read the docstring.

Reply:
done

Reply:
I don't think you addressed this. Did you look up what a context manager class is?

At the very least, the docstring should explain that this is a context manager class, as I suggested in my previous review comment.